pkg/trait/knative.go

```diff
@@ -486,9 +488,18 @@ func (t *knativeTrait) createTrigger(e *Environment, ref *corev1.ObjectReference
 		if ref.Namespace == "" {
 			ref.Namespace = e.Integration.Namespace
 		}
-		trigger := knativeutil.CreateTrigger(*ref, e.Integration.Name, eventType, path)
+		serviceAPIVersion, err := t.determineServiceType(e)
+		if err != nil {
+			return err
+		}
+		trigger, err := knativeutil.CreateTrigger(*ref, e.Integration.Name, serviceAPIVersion, eventType, path)
```

Comment:
In order to simplify the maintenance, see the some comment below related to the CreateTrigger func. I think here, we should just call either CreateTriggerService or CreateTriggerKnativeService to make it more explicit. Also mind that we do have CronJob controller strategy.